client/src/app/site/polls/services/poll.service.ts

```diff
@@ -113,23 +113,42 @@ export class PollService {
             .subscribe(isEnabled => (this.isElectronicVotingEnabled = isEnabled));
     }
 
+    /**
+     * Compares two numbers.
+     * @param a the first number
+     * @param b the number that needs to be compared with a
+     * @returns 0 if they are equal, a negative value if a>b, else a positive value
+     */
+    private compareValues(a: number, b: number): number {
+        if (!b) {
+            if (!a) {
+                return 0;
+            }
+            return -1;
+        }
+        if (!a) {
+            return 1;
+        }
+        return b - a;
+    }
```

Comment:
Please make it a more re-usable/generic function and extract it into an own file.